lib/glimesh/account_follows.ex

```diff
@@ -18,49 +18,52 @@ defmodule Glimesh.AccountFollows do
 
   defp sub_and_return(topic), do: {Phoenix.PubSub.subscribe(Glimesh.PubSub, topic), topic}
 
-  defp broadcast({:error, _reason} = error, _event), do: error
-
-  defp broadcast({:ok, %Follower{} = following}, :followers = event) do
+  defp broadcast(%Follower{} = following, :followers = event) do
     Glimesh.Events.broadcast(
       get_subscribe_topic(:follows, following.streamer_id),
       get_subscribe_topic(:follows),
       event,
       following
     )
-
-    {:ok, following}
   end
 
   def follow(%User{} = streamer, %User{} = user, live_notifications \\ false) do
-    attrs = %{
-      has_live_notifications: live_notifications
-    }
 
-    results =
+    result =
       %Follower{
         streamer: streamer,
-        user: user
+        user: user,
+        has_live_notifications: live_notifications
       }
-      |> Follower.changeset(attrs)
+      |> Follower.changeset()
```

Comment:
Not a big deal but I think setting values when creating the object is cleaner and has better type checking than passing in attributes.

Converting the object to a changeset is needed to add constraint checks still, otherwise an insert with an existing row will throw.

Reply:
The reason `changeset/2` is used is to actually validate that we allow changes to the property, and that it's being changed into a valid type. Take for example if `Follower` had `validate_subset(attrs, :has_live_notifications, [true, false])` inside it's changeset, then we could get the following benefit:
```elixir
iex(9)> %Glimesh.AccountFollows.Follower{has_live_notifications: "foobar"} |> Glimesh.AccountFollows.Follower.changeset(%{})
#Ecto.Changeset<
  action: nil,
  changes: %{},
  errors: [
    streamer: {"can't be blank", [validation: :required]},
    user: {"can't be blank", [validation: :required]}
  ],
  data: #Glimesh.AccountFollows.Follower<>,
  valid?: false
>

iex(11)> %Glimesh.AccountFollows.Follower{} |> Glimesh.AccountFollows.Follower.changeset(%{has_live_notifications: "foobar"})
#Ecto.Changeset<
  action: nil,
  changes: %{},
  errors: [
    streamer: {"can't be blank", [validation: :required]},
    user: {"can't be blank", [validation: :required]},
    has_live_notifications: {"is invalid", [type: :boolean, validation: :cast]}
  ],
  data: #Glimesh.AccountFollows.Follower<>,
  valid?: false
>
```

You're right though it has no benefit here, except for being able to be reused if for example a user has access to flip this raw data at some point.

Reply:
Ah neat, I didn't realize that thanks. Switched back `changeset/2` as it was before.